### DISCUSSION TITLE
Issue enabling AC and AA

### DISCUSSION THREAD

**Discussion by lucas14231** [Mr-MIBonk/M.I.B._More-Incredible-Bash#160] [Q&A]
Hi

I've been trying to get carplay and android auto working on my 2017 audi tt for a bit now

After following forums, I updated the firmware of the car and installed more incredible bash, I ran the AIO patch, re-booted, and the Audi smartphone interface logo appeared within the menu. However, it isn't functional and doesn't recognise any phones. I've tried a few phones with different cables also.

I've checked the USB ports, I am able to play music through a pen drive, so I know that they aren't for charging only

After reformatting my SD card and re-copying MIB to it, I went ahead and re-did the AIO patch which is supposed to add the correct FEC codes into the FEC list (60800 and 60900 needed). The SD card is a Sandisk ultra 32GB.

After this, I re-booted the unit and had a look at the activation keys within the red engineering menu, the codes still aren't listed:

![PXL_20230325_161523497](https://user-images.githubusercontent.com/128925023/227732260-5f4f4890-4ba8-4d37-a928-f9a682930df3.jpg)

However, at the end of the installation process for the AIO patch, I recieved the following info:

![PXL_20230325_16323052045](https://user-images.githubusercontent.com/128925023/227732104-c5e75069-cf32-4b0d-bb0d-90bc1424de81.jpg)

When I got the car a few weeks ago, after some research I updated the firmware to TT_MHI2_ER_AU43x_P5098

I'm guessing that re-installing the stock firmware and re-trying could be the way to go? Would just re-updating the firmware via the REM be the best way to do this? Or is there something within MIB that would allow me to revert back to stock firmware

Any help is appreciated

Lucas

**Comment by harman-f** [comment]
Log file would be helpful.

What is the PN of your cluster and which FW is installed?

**Reply by lucas14231** [reply]
[MU1339-JTH-07614.05.1734090086[3096].log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/11069695/MU1339-JTH-07614.05.1734090086.3096.log)

Here's a log file

![PXL_20230325_175840693](https://user-images.githubusercontent.com/128925023/227733903-850fe9cf-5bf1-4e1e-985b-eee8e3b045d6.jpg)

And heres a pic of the version screen, with 1339 installed

Sorry but not sure what you mean by PN

**Comment by harman-f** [comment]
have you copied the patch to the right folder on SD?

**Reply by lucas14231** [reply]
Yeah that's sorted it

After downloading the patch files from https://mibsolution.one/wl/?id=ct7tZHJ2XdRwxWAhs1cHmPMm8pk1YYGa and copying the one related to my FW version to the patches folder on the SD card (MHI2_ER_AU43x_P5098_MU1339_PATCH), I ran the AIO process again and boom, all good and working. Teaches me to not skip over stuff in guides

![PXL_20230325_195102762](https://user-images.githubusercontent.com/128925023/227739201-4d8529db-4931-4fcb-85a6-9c54439d4499.jpg)

Thanks a lot for recognising the problem for me

**Comment by harlem1992** [comment]
__  __     ___           ___  
|  \/  |   |_ _|         | _ ) 
| |\/| |    | |          | _ \ 
|_|  |_|ore|___|ncredible|___/ash - pre-release V3.7.1 "MHIG Edition"
______________________________________________________________________

NOT FOR COMMERCIAL USE - IF YOU BOUGHT THIS YOU GOT RIPPED OFF

Logging_MU1502-MHI2_JP_POG11_K1503-AW7-MON07.11.1800060978:
PN: 9P1035897K - HW Number: 055
Metainfo2 Variant (Variant2): FM2-P-TNL-JP-PO-MLP
EEPROM Variant (Variant1): MU-P-LN-JP
Parameter set part number: -----------
Coding: 07760406be0a000003330001061200102f0003c6037041065e
Car Model: Porsche 911 Carerra Coupe
VIN MIB:  - VIN car: 
______________________________________________________________________

ls --> total 2920
      64 drwxrwxrwx 11 root      root          32768 Jan 01  1970 .
      64 drwxrwxrwx 11 root      root          32768 Jan 01  1970 ..
      64 drwxrwxrwx  2 root      root          32768 Apr 16 21:09 .System Volume Information
       2 -rwxrwxrwx  1 root      root            986 Apr 16 12:54 .git-blame-ignore-revs
       1 -rwxrwxrwx  1 root      root            399 Apr 16 12:54 .gitattributes
      64 drwxrwxrwx  3 root      root          32768 Apr 16 18:32 .github
       1 -rwxrwxrwx  1 root      root            132 Apr 16 12:54 .gitignore
       0 n---------  1 root      root              0 Apr 16 21:09 =64 GB
      81 -rwxrwxrwx  1 root      root          41255 Apr 16 12:54 GEM.png
      30 -rwxrwxrwx  1 root      root          14939 Apr 16 12:54 LICENSE
      64 drwxrwxrwx  4 root      root          32768 Apr 16 18:30 Launcher
       1 -rwxrwxrwx  1 root      root             60 Apr 16 12:54 M.I.B. - More Incredible Bash - Mibsolution Wiki.url
       0 -rwxrwxrwx  1 root      root              0 Apr 16 12:54 NOT FOR COMMERCIAL USE - IF YOU BOUGHT THIS YOU GOT RIPPED OFF.txt
     284 -rwxrwxrwx  1 root      root         144962 Apr 16 12:54 PATCH COMPATIBILITY TABLE.pdf
    1703 -rwxrwxrwx  1 root      root         871810 Apr 16 12:54 Password_List_V4.0.pdf
      13 -rwxrwxrwx  1 root      root           6507 Apr 16 12:54 README.md
       0 -rwxrwxrwx  1 root      root              0 Apr 16 12:54 Swdlautorun.txt
       1 -rwxrwxrwx  1 root      root             34 Apr 16 12:54 VERSION
       1 -rwxrwxrwx  1 root      root            296 Jan 01  1980 WMPInfo.xml
      64 drwxrwxrwx  3 root      root          32768 Apr 16 18:32 apps
      64 drwxrwxrwx  3 root      root          32768 Jan 01  1980 backup
      77 -rwxrwxrwx  1 root      root          39351 Apr 16 12:54 bash.png
      64 drwxrwxrwx  2 root      root          32768 Apr 16 18:30 config
      64 drwxrwxrwx  3 root      root          32768 Apr 16 18:30 esd
       5 -rwxrwxrwx  1 root      root           2057 Apr 16 12:54 metainfo2.txt
      64 drwxrwxrwx 13 root      root          32768 Apr 16 18:30 mod
      64 drwxrwxrwx 188 root      root          32768 Apr 16 18:32 patches
      16 -rwxrwxrwx  1 root      root           7874 Apr 16 12:54 start
______________________________________________________________________

timestamp: 2024_04_16_21_14_18----->

RCC-showimage---->
No end time for image set
--- Stop image display with -unload

--- Running basic backup --------------------

timestamp: 2024_04_16_21_14_23----->

RCC-backup---->
Found AW7-MON07.11.1800060978 backup, will check if all complete...
Doing basic backup
It will take about 2 minutes to complete.
Please wait...
Version info.....
timestamp: 2024_04_16_21_14_28----->

RCC-allversions---->
No info available in /net/mmx/mnt/app/navigation/navStartup.
'/mnt/app/wordprediction/awp' is not an executable file in PATH: ':/proc/boot:/sbin:/bin:/usr/bin:/usr/sbin:/net/mmx/bin:/net/mmx/usr/bin:/net/mmx/usr/sbin:/net/mmx/sbin:/net/mmx/mnt/app/armle/bin:/net/mmx/mnt/app/armle/sbin:/net/mmx/mnt/app/armle/usr/bin:/net/mmx/mnt/app/armle/usr/sbin:/net/mmx/mnt/app/root/lib-target:/net/mmx/mnt/eso/lib:/net/mmx/eso/lib:/net/mmx/mnt/app/usr/lib:/net/mmx/mnt/app/armle/lib:/net/mmx/mnt/app/armle/lib/dll:/net/mmx/mnt/app/armle/usr/lib'.
 -> done :-)
FEC folders...... -> done :-)
Shadow files..... -> done :-)
Partition info... -> done :-)
Folder lists..... -> done :-)
Datasets......... -> done :-)
EEPROM........... -> done :-)
storage1......... -> done :-)
storage2......... -> done :-)
backing up backup -> part 1 done
ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01  1970 .
      64 drwxrwxr-x  7 root      root          32768 Jan 01  1970 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01  1970 MU1502
Free space in /net/mmx/mnt/boardbook 973Mbytes

RCC-backup-done!

--- Running Launcher script -----------------

timestamp: 2024_04_16_21_16_19----->

RCC-launcher---->
-- Swdlautorun.txt disabled
-- Skip MOST popup enabled
-- Illegal and withdrawn FECs removed
-- Cleanup done!
-- AppDevelopment.jar found
-- GEM update is skipped (sha1=4826f6b657faa144eacc5ddaecb4dd5430eb8f46)
-- MHI2 esd installed
-- M.I.B GEM link installed
-- activating Development Mode (GEM)...

--- Unload PNG ------------------------------

timestamp: 2024_04_16_21_16_30----->

RCC-showimage---->
--- Stopping image display
[stuffScript] Cool stuff DONE

timestamp: 1970_01_01_00_05_33----->

MMX-carp---->
CarPlay and Android Auto activated now

MMX-carp---->
Reboot Unit...
Unit is rebooting now...

timestamp: 2024_04_16_21_34_58----->

RCC-addimage---->
Installation finished :)

RCC-addimage---->
Reboot Unit...
Unit is rebooting now...

timestamp: 1970_01_01_00_02_08----->

MMX-carp---->
CarPlay and Android Auto activated now

MMX-carp---->
Reboot Unit...
Unit is rebooting now...

timestamp: 2024_04_16_22_53_00----->

RCC-carp---->
CarPlay and Android Auto activated now

timestamp: 1970_01_01_00_05_47----->

MMX-addimage---->
File checksum does not match. No patch applied
Button images are already installed

timestamp: 1970_01_01_00_05_51----->

MMX-flash---->

MMX-flash---->
ifs-root-stage2.ifs offset: 0x00ba0000
ifs-root-stage2.ifs size: 0C05F400
Found AW7-MON07.11.1800060978 backup, will check if all complete...
Doing standard backup
It will take about 5 minutes to complete.
Please wait...
Version info..... -> Backup is already there!
FEC folders...... -> Backup is already there!
Shadow files..... -> Backup is already there!
Partition info... -> Backup is already there!
Folder lists..... -> Backup is already there!
Datasets......... -> Backup is already there!
EEPROM........... -> Backup is already there!
storage1......... -> Backup is already there!
storage2......... -> Backup is already there!
SWDL............. -> done :-)
GEM jars......... -> done :-)
lsd.jxe.......... -> done :-)
HMI folders...... -> done :-)
RCC OFFSET's..... -> done :-)
RCC Flashlock.... -> done :-)
RCC flash dump...
timestamp: 1970_01_01_00_06_45----->

MMX-addfec---->
--- Stock FECs from /net/rcc/mnt/efs-system/backup/FEC/FecContainer.fec
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00

--- FECs in use from /net/rcc/mnt/efs-persist/FEC/FecContainer.fec:
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00


timestamp: 1970_01_01_00_06_51----->

MMX-fecel---->
--- /patches/MHI2_JP_POG11_K1503_MU1502_PATCH/addFecs.txt not found!
Stopping here....
 -> done :-)
RCC ifs-stage2...
timestamp: 1970_01_01_00_06_55----->

MMX-fecel---->
--- Checking if stock EL is installed
    Stock EL found on unit
4512+0 records in
4512+0 records out
 -> done :-)
MMX flash dump...
timestamp: 1970_01_01_00_07_00----->

MMX-addfec---->
--- Stock FECs from /net/rcc/mnt/efs-system/backup/FEC/FecContainer.fec
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00

--- FECs in use from /net/rcc/mnt/efs-persist/FEC/FecContainer.fec:
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00

 -> done :-)
backing up backup -> Backup part 1 is already there!
ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:02 .
      64 drwxrwxr-x  7 root      root          32768 Jan 01 00:02 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:07 MU1502
Free space in /net/mmx/mnt/boardbook 845Mbytes

MMX-flash-done!

Patch folder /net/mmx/fs/sda0/patches/MHI2_JP_POG11_K1503_MU1502_PATCH NOT found on SD card

--------------- DEBUG --------------
IFS_VERSION:		
OFFSETPART2:		ba0000
OFFSETPART2_DB:		
STAGE2SIZE:		0C05F400
STAGE2SIZE_orig:	
SHA1_orig:		
header_chk_FEC_CP:	
SHA1_FEC_CP:		
IFS2_new:		
IFS2_backup:		
Big Endian:		
SHA1_UNIT:		
SHA1_PATCH:		
------------------------------------

timestamp: 2024_04_16_22_55_10----->

RCC-addfec---->
--- Stock FECs from /net/rcc/mnt/efs-system/backup/FEC/FecContainer.fec
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00

--- FECs in use from /net/rcc/mnt/efs-persist/FEC/FecContainer.fec:
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00


timestamp: 1970_01_01_00_11_38----->

MMX-wlan---->
WLAN is activated now
07760406be0a000003330001061200102f0003c6037041065e

MMX-wlan---->
Reboot Unit...
Unit is rebooting now...

timestamp: 2024_04_16_23_10_18----->

RCC-carp---->
CarPlay and Android Auto activated now

timestamp: 1970_01_01_00_10_55----->

MMX-addimage---->
File checksum does not match. No patch applied
Button images are already installed

timestamp: 1970_01_01_00_11_00----->

MMX-flash---->

MMX-flash---->
ifs-root-stage2.ifs offset: 0x00ba0000
ifs-root-stage2.ifs size: 0C05F400
Found AW7-MON07.11.1800060978 backup, will check if all complete...
Doing standard backup
It will take about 5 minutes to complete.
Please wait...
Version info..... -> Backup is already there!
FEC folders...... -> Backup is already there!
Shadow files..... -> Backup is already there!
Partition info... -> Backup is already there!
Folder lists..... -> Backup is already there!
Datasets......... -> Backup is already there!
EEPROM........... -> Backup is already there!
storage1......... -> Backup is already there!
storage2......... -> Backup is already there!
SWDL............. -> Backup is already there!
GEM jars......... -> Backup is already there!
lsd.jxe.......... -> Backup is already there!
HMI folders...... -> Backup is already there!
RCC OFFSET's..... -> Backup is already there!
RCC Flashlock.... -> Backup is already there!
RCC flash dump... -> Backup is already there!
RCC ifs-stage2... -> Backup is already there!
MMX flash dump... -> Backup is already there!
backing up backup -> Backup part 1 is already there! -> Backup part 2 is already there!

ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:02 .
      64 drwxrwxr-x  7 root      root          32768 Jan 01 00:02 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:07 MU1502
Free space in /net/mmx/mnt/boardbook 845Mbytes

MMX-flash-done!

Patch folder /net/mmx/fs/sda0/patches/MHI2_JP_POG11_K1503_MU1502_PATCH NOT found on SD card

--------------- DEBUG --------------
IFS_VERSION:		
OFFSETPART2:		ba0000
OFFSETPART2_DB:		
STAGE2SIZE:		0C05F400
STAGE2SIZE_orig:	
SHA1_orig:		
header_chk_FEC_CP:	
SHA1_FEC_CP:		
IFS2_new:		
IFS2_backup:		
Big Endian:		
SHA1_UNIT:		
SHA1_PATCH:		
------------------------------------

timestamp: 1970_01_01_00_11_32----->

MMX-addfec---->
--- Stock FECs from /net/rcc/mnt/efs-system/backup/FEC/FecContainer.fec
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00

--- FECs in use from /net/rcc/mnt/efs-persist/FEC/FecContainer.fec:
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00


timestamp: 1970_01_01_00_11_37----->

MMX-fecel---->
--- /patches/MHI2_JP_POG11_K1503_MU1502_PATCH/addFecs.txt not found!
Stopping here....

timestamp: 1970_01_01_00_11_40----->

MMX-fecel---->
--- Checking if stock EL is installed
    Stock EL found on unit

timestamp: 1970_01_01_00_11_45----->

MMX-addfec---->
--- Stock FECs from /net/rcc/mnt/efs-system/backup/FEC/FecContainer.fec
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00

--- FECs in use from /net/rcc/mnt/efs-persist/FEC/FecContainer.fec:
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00


timestamp: 1970_01_01_00_15_01----->

MMX-addfec---->
--- Stock FECs from /net/rcc/mnt/efs-system/backup/FEC/FecContainer.fec
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00

--- FECs in use from /net/rcc/mnt/efs-persist/FEC/FecContainer.fec:
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00


timestamp: 1970_01_01_00_15_06----->

MMX-fecel---->
--- /patches/MHI2_JP_POG11_K1503_MU1502_PATCH/addFecs.txt not found!
Stopping here....

timestamp: 1970_01_01_00_15_09----->

MMX-fecel---->
--- Checking if stock EL is installed
    Stock EL found on unit

timestamp: 1970_01_01_00_15_14----->

MMX-addfec---->
--- Stock FECs from /net/rcc/mnt/efs-system/backup/FEC/FecContainer.fec
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00

--- FECs in use from /net/rcc/mnt/efs-persist/FEC/FecContainer.fec:
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00


timestamp: 2024_04_16_23_15_00----->

RCC-addfec---->
--- Stock FECs from /net/rcc/mnt/efs-system/backup/FEC/FecContainer.fec
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00

--- FECs in use from /net/rcc/mnt/efs-persist/FEC/FecContainer.fec:
00060500
00070200
00070100
00040100
06330028
00030000
00050000
00060700
00060800
00060100
00060A00

**Reply by harlem1992** [reply]
Hi, Ive been trying to get my android auto working for the following patch MHI2_JP_POG11_K1503 but it connects and ask to allow for android auto but nothing happens after that. Apple Carplay works seemlessly for this before intallling MIB. 

I saw an option to convert to EU, should i do that or better to just leave it as it is.
